### PR TITLE
Detach and archive old partitions

### PR DESCRIPTION
**Pull request by conbrad**
Every month, create and attach a new partition for the next month and attempt to detach and archive partitions that are 3 to 6 months old.
# Test Links:
[Landing Page](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-4138-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: openshift/pgslice/docker/partition_and_archive.sh [hunk 1]
```diff
@@ -51,24 +51,57 @@ fi
...
-psql -c "DROP TABLE ${TABLE}_retired" $PGSLICE_URL
\ No newline at end of file
+
+# Add new partition for next month
+NEXT_MONTH_DATE=$(date -d "$(date +%Y-%m-01) next month" +%Y%m)
+FIRST_DAY_NEXT_MONTH=$(date -d "$(date +%Y-%m-01) next month" +%Y-%m-%d)
+LAST_DAY_NEXT_MONTH=$(date -d "$(date +%Y-%m-01) next month +1 month -1 day" +%Y-%m-%d)
+echo "Creating new partition for dates: $FIRST_DAY_NEXT_MONTH to $LAST_DAY_NEXT_MONTH"
+
+NEW_PARTITION="${TABLE}_${NEXT_MONTH_DATE}"
```
Comment (dgboss) [inline openshift/pgslice/docker/partition_and_archive.sh:61]: `NEW_PARTITION` is unused

File: openshift/pgslice/docker/partition_and_archive.sh [hunk 2]
```diff
@@ -51,24 +51,57 @@ fi
...
+DETACH_COMMAND="
+DO $$ 
+BEGIN
+    IF EXISTS (SELECT 1 FROM pg_catalog.pg_partitions WHERE schemaname = 'public' AND tablename = 'weather_station_model_predictions_202408') THEN
```
Comment (dgboss) [inline openshift/pgslice/docker/partition_and_archive.sh:65]: I assume the table name will be parameterized before deployment!

Comment (dgboss) [reply openshift/pgslice/docker/partition_and_archive.sh:65]: Oh, was this first 'detach' block just for testing?

Comment (conbrad) [reply openshift/pgslice/docker/partition_and_archive.sh]: Yep, removed in: https://github.com/bcgov/wps/pull/4138/commits/828820760efec5e0ac4156525458f55351c35183

**Review by dgboss — APPROVED** [review]
Nice work!

**Pull request merged**